theano/sandbox/cuda/basic_ops.py

```diff
@@ -3044,6 +3044,28 @@ def make_node(self, *axis_and_tensors):
         as_tensor_variable_args = [as_cuda_ndarray_variable(x)
                                    for x in tensors]
 
+        # Get joining axis as int
+        axis_int = 0
+        if not isinstance(axis, int):
+            try:
+                # Note : `get_scalar_constant_value` returns a ndarray not
+                # an int
+                axis_int = int(tensor.get_scalar_constant_value(axis))
+
+            except tensor.basic.NotScalarConstantError:
+                pass
+	else:
+	    axis_int = axis
+
+        if (axis_int < 0):
+            # Find tensor with smallest number of dimensions
+            min_dim = -1
```

Comment:
I agree general code could be better when well planed. However, in this case I do not see how we could support different number of dimensions, how would you concatenate this? Numpy, for instance, does not ([see doc](http://docs.scipy.org/doc/numpy/reference/generated/numpy.concatenate.html)) support it and theano's CPU version is based on numpy implementation.